### PR TITLE
Jobs manage

### PR DESCRIPTION
**Pull request by ea-rus**
## Description

Please include a summary of the change and the issue it solves. 

[DOCS](https://github.com/mindsdb/mindsdb/blob/jobs-manage/mindsdb/interfaces/jobs/README.md)

https://www.loom.com/share/fda7d7ada2a14a4da83f380b956805cb

## Type of change

(Please delete options that are not relevant)

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📄 This change requires a documentation update

### What is the solution?

(Describe at a high level how the feature was implemented)

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation, or created issues to update them.
- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] I have shared a short loom video or screenshots demonstrating any new functionality.

### CODE REVIEW

File: mindsdb/interfaces/jobs/README.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+)
+START { now | <date> }
+END { now | <date> }
+EVERY [<numer>] <period> 
+```
+Parameters:
+- START - optional, first date when job have to start. Default is now
+- END - optional, after this date job shouldn't run anymore. If not set - repeat forever
+- EVERY - optional job repeat rule.
+  - If not set - run once (no mather what was set in END)
```
Comment (paxcema) [inline mindsdb/interfaces/jobs/README.md:17]: typo - ~mather~ -> matter

File: mindsdb/interfaces/jobs/README.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+```
+
+This example creates job in 'proj1' project and runs it every 2 days till 1st April. 
+
+This job have to
+- retrain predictor 
+- use trained predictor on records from table 'int2.table1'
+- append results of prediction to int1.rentals table
+
+join_learn_process=true - this parameter is used for waiting to retrain predictor
```
Comment (paxcema) [inline mindsdb/interfaces/jobs/README.md:55]: Thinking at a higher level, maybe it's worth introducing a blocking mechanism of some kind for this? Because other jobs like adjust or create would also be async and hence we need to wait. Or maybe just using `join_learn_process` on all three of those is good enough for now.

File: mindsdb/interfaces/jobs/jobs_controller.py [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+import sqlalchemy as sa
+
+from mindsdb_sql import parse_sql, ParsingException
+
+from mindsdb.interfaces.storage import db
+from mindsdb.interfaces.database.projects import ProjectController
+from mindsdb.utilities.context import context as ctx
+from mindsdb.utilities import log
+
+mp_ctx = mp.get_context('spawn')
```
Comment (paxcema) [inline mindsdb/interfaces/jobs/jobs_controller.py:15]: Shouldn't this be inside a try-except? I think the main entry point to mindsdb is already setting this.

File: mindsdb/interfaces/jobs/scheduler.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+import torch.multiprocessing as mp
+
+from mindsdb.utilities.config import Config
+from mindsdb.utilities.log import initialize_log
+from mindsdb.utilities import log
+from mindsdb.interfaces.storage import db
+
+from mindsdb.interfaces.jobs.jobs_controller import JobsExecutor
+
+mp_ctx = mp.get_context('spawn')
```
Comment (paxcema) [inline mindsdb/interfaces/jobs/scheduler.py:13]: Same as the comment above.

File: mindsdb/interfaces/jobs/scheduler.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        logger.debug('Scheduler check timetable')
+        try:
+            check_timetable(config)
+        except (SystemExit, KeyboardInterrupt):
+            raise
+        except Exception as e:
+            logger.error(e)
+
+        # different instances should start in not the same time
+        time.sleep(20 + random.randint(1, 10))
```
Comment (paxcema) [inline mindsdb/interfaces/jobs/scheduler.py:30]: Shouldn't this also be configurable? I can envision local deployments where the user wants monitoring at the per-second level (or maybe fractions of a second!).

File: mindsdb/interfaces/jobs/scheduler.py [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+def execute_sync(record, config):
+    logger.info(f'Job execute: {record.name}({record.id})')
+
+    exec_method = config.get('jobs', {}).get('executor', 'local')
+
+    task_process(record.id, exec_method)
+
+
+def execute_async(record, config):
+    # if async is used with training in background then training is not happening
```
Comment (paxcema) [inline mindsdb/interfaces/jobs/scheduler.py:49]: Should we check this?

For example, when parsing the query if would be invalid to ask for a train/retrain/adjust job if there is no `USING` clause with `join_learn_process` set to True.

**Review by paxcema — CHANGES_REQUESTED** [review]
Overall looks good to me! Left a couple of suggestions but maybe those are best left for another PR.

File: mindsdb/interfaces/jobs/README.md (hunk 2 above)
Comment (ea-rus) [reply mindsdb/interfaces/jobs/README.md:55]: I think join_learn_process is enough for now. Because it should work for retrain and adjust too

File: mindsdb/interfaces/jobs/jobs_controller.py (hunk 3 above)
Comment (ea-rus) [reply mindsdb/interfaces/jobs/jobs_controller.py:15]: I looked at the other examples of mp.get_context, they are not try-catched

File: mindsdb/interfaces/jobs/scheduler.py (hunk 5 above)
Comment (ea-rus) [reply mindsdb/interfaces/jobs/scheduler.py:30]: I will add to config. But for now we can't run job more often than every minute. So precision is not about seconds now.

File: mindsdb/interfaces/jobs/README.md (hunk 2 above)
Comment (paxcema) [reply mindsdb/interfaces/jobs/README.md:55]: Agreed 👍

File: mindsdb/interfaces/jobs/jobs_controller.py (hunk 3 above)
Comment (paxcema) [reply mindsdb/interfaces/jobs/jobs_controller.py]: My bad, I was thinking of `set_start_method`, ignore this)

File: mindsdb/interfaces/jobs/scheduler.py (hunk 6 above)
Comment (paxcema) [reply mindsdb/interfaces/jobs/scheduler.py:52]: Makes sense

**Review by paxcema — APPROVED** [review]
